app/src/main/java/com/openlattice/chronicle/services/status/ParticipationStatusMonitoringService.kt

```diff
@@ -0,0 +1,82 @@
+import android.util.Log
+import com.crashlytics.android.Crashlytics
+import com.openlattice.chronicle.ChronicleStudyApi
+import com.openlattice.chronicle.data.ParticipationStatus
+import com.openlattice.chronicle.preferences.EnrollmentSettings
+import com.openlattice.chronicle.services.upload.PRODUCTION
+import com.openlattice.chronicle.services.upload.cancelUploadJobScheduler
+import com.openlattice.chronicle.services.upload.createRetrofitAdapter
+import com.openlattice.chronicle.services.upload.scheduleUploadJob
+import com.openlattice.chronicle.services.usage.cancelUsageMonitoringJobScheduler
+import com.openlattice.chronicle.services.usage.scheduleUsageMonitoringJob
+import io.fabric.sdk.android.Fabric
+import java.lang.Exception
+import java.util.*
+import java.util.concurrent.Executors
+
+const val STATUS_CHECK_PERIOD_MILLIS = 15 * 60 * 1000L
+
+class ParticipationStatusMonitoringService : JobService() {
+    private val executor = Executors.newSingleThreadExecutor()
+    private val chronicleStudyApi = createRetrofitAdapter(PRODUCTION).create(ChronicleStudyApi::class.java)
+
+    override fun onCreate() {
+        super.onCreate()
+        Fabric.with(this, Crashlytics())
+    }
+    override fun onStopJob(p0: JobParameters?): Boolean {
+        Log.i(javaClass.name, "Participation status service stopped")
+        executor.shutdown()
+        return true // reschedule the job
+    }
+
+    override fun onStartJob(parameters: JobParameters?): Boolean {
+        Log.i(javaClass.name, "Participation status service initialized")
+
+        val enrollmentSettings = EnrollmentSettings(applicationContext)
+        val studyId :UUID = enrollmentSettings.getStudyId()
+        val participantId :String = enrollmentSettings.getParticipantId()
+
+        executor.execute {
+            var participationStatus = enrollmentSettings.getParticipationStatus()
+            try {
+                participationStatus = chronicleStudyApi.getParticipationStatus(studyId, participantId)
+            } catch (e :Exception) {
+                Crashlytics.log("caught exception: studyId: \"$studyId\" participantId: \"$participantId\"")
+                Crashlytics.logException(e)
+            }
+            Log.i(javaClass.name, "Participation status: $participationStatus")
+
+            if (participationStatus == ParticipationStatus.ENROLLED) {
+                scheduleUploadJob(this)
+                scheduleUsageMonitoringJob(this)
+            } else {
+                cancelUsageMonitoringJobScheduler(this)
+                cancelUploadJobScheduler(this)
+            }
+
+            enrollmentSettings.setParticipationStatus(participationStatus)
+            jobFinished(parameters, false)
+        }
+        return true
+    }
+}
+
+fun scheduleParticipationStatusJob(context :Context) {
+    val serviceComponent = ComponentName(context, ParticipationStatusMonitoringService::class.java)
+    val jobBuilder = JobInfo.Builder(0, serviceComponent)
```

Comment:
In some places, you use a constant for the job id, but not here. 

```
JobInfo.Builder(UPLOAD_JOB_ID...
JobInfo.Builder(MONITOR_USAGE_JOB...
```

Assuming these job ids must be unique, I recommend creating a constants file to hold all the job ids in one place to ensure they are unique, something like...

```
// Jobs.kt
enum class Jobs(val id: Int) {
    UPLOAD_JOB_ID(0),
    MONITOR_USAGE_JOB(1),
    MONITOR_PARTICIPATION_STATUS_JOB(2)
}

```